### ISSUE TITLE
Audio component does not record for users on iOS 

### ISSUE THREAD

**Issue by abidlabs** [gradio-app/gradio#2646]
### Describe the bug

We've been getting reports that the Microphone does not seem to work for users on iOS, even once they provide permissions. Here are some of the reports:

* https://twitter.com/ManuelFaysse/status/1592125388452691968
* https://huggingface.co/spaces/fffiloni/whisper-to-stable-diffusion/discussions/4

I've also confirmed I see the problem with a simple repro:

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Try running this:

```py
import gradio as gr

gr.Interface(lambda x:x, "mic", "audio").launch(share=True)
```

And open the share link on an iOS device

### Logs

```shell
Hard to see the logs since it's on iOS
```

### System Info

```shell
gradio==3.9.1
```

### Severity

serious, but I can work around it

**Comment by fffiloni** [comment]
Recording works fine i guess, as we can listen to the recorded audio from the html5 player, but it may be a format/codec issue from iOS 🤗

**Comment by MaxBrodeurUrbas** [comment]
+1 for this issue. Audio recording fails consistently on iOS and a user of our gradio demo also reported the same issue when trying the demo on a macbook + chrome.

**Comment by KamyMouss** [comment]
+1, doesn't work on my end either. Would appreciate a fix asap!

**Comment by aliabid94** [comment]
Taking a look!

**Comment by leobaumgardt** [comment]
+1.

Also, in my case, the recorded audio is silent. That is to say, if I record for 5 seconds, I will get a player with 5 seconds of recorded audio. But when I play it, I will hear nothing.

This is using iOS Safari via the "share" URL.

I tested record/playback on another site (https://addpipe.com/simple-web-audio-recorder-demo/), and there it works.

Perhaps it has something to do with iOS using "audio/webm"?

**Comment by MaxBrodeurUrbas** [comment]
I've also had users report audio recording completely fails on older versions of MacOS (Catalina 10.15). Failing occurs regardless of browser and selecting example audio files from the gradio.example list is also broken. Not sure if this is relevant to the current bug.

**Comment by abidlabs** [comment]
This should be solved soon with #2961, thanks @aliabid94!
